onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h

```diff
@@ -7,12 +7,14 @@
 #include "core/framework/op_kernel.h"
 #include "NvInfer.h"
 #include "NvOnnxParser.h"
+#include "core/platform/ort_mutex.h"
 
 namespace onnxruntime {
 
 static const int kMaxBatchSize = 1;
-static const int kMaxWorkSpaceSize = 1 << 30;
-
+static const int kMaxWorkSpaceSize = 1 << 30;    
+static OrtMutex tensorrt_mu;
```

Comment:
better if we don't use global statics.
can't the mutex be part of the provider? and a reference kept in the compute state?